### PR TITLE
Rewrite PodSecurityPolicy migration guide

### PR DESCRIPTION
**Pull request by tallclair**
Add a lot more detail to the PSP migration guide. Notable changes include:

- Detailed step-by-step guide for how to migrate PSP resources
- Additional detail around permissions & policies for namespaces
- Detailed table of PSP fields, with mutating/validating type and PSA equivalents

Fixes https://github.com/kubernetes/website/issues/31367

As a side note, if anyone is interested in building some tooling to assist with the migration, the most manual steps are:

- Verification of mutating pods (step 2c)
- PSP simplification steps, in general
- RBAC review tooling (who can relabel namespaces, which PSPs are bound to a namespace, etc.)

/kind documentation
/sig auth security
/assign @liggitt @tabbysable @sftim @photoni-137

### CODE REVIEW

File: content/en/docs/tasks/configure-pod-container/migrate-from-psp.md
Comment (tengqm) [inline content/en/docs/tasks/configure-pod-container/migrate-from-psp.md:123]: What does "Pod Security" stand for here? The "Pod Security Admission" or "PodSecurityPolicy"?

File: content/en/docs/tasks/configure-pod-container/migrate-from-psp.md
Comment (tengqm) [inline content/en/docs/tasks/configure-pod-container/migrate-from-psp.md:150]:
```suggestion
- `.spec.runAsUser` - (Non-mutating with the `RunAsAny` rule) required to enforce `RunAsNonRoot` for
```

File: content/en/docs/tasks/configure-pod-container/migrate-from-psp.md
Comment (tengqm) [inline content/en/docs/tasks/configure-pod-container/migrate-from-psp.md:189]:
```suggestion
   the original PodSecurityPolicies. If the pods haven't been updated since the new policies were rolled out,
```

File: content/en/docs/tasks/configure-pod-container/migrate-from-psp.md
Comment (tengqm) [inline content/en/docs/tasks/configure-pod-container/migrate-from-psp.md:202]:
```suggestion
familiarizing yourself with the 3 different Pod security levels.
```

File: content/en/docs/tasks/configure-pod-container/migrate-from-psp.md
Comment (tengqm) [inline content/en/docs/tasks/configure-pod-container/migrate-from-psp.md:206]:
```suggestion
1. **By security requirements for the namespace** - If you are familiar with the expected access
```

File: content/en/docs/tasks/configure-pod-container/migrate-from-psp.md
Comment (tengqm) [inline content/en/docs/tasks/configure-pod-container/migrate-from-psp.md:210]:
```suggestion
   PSP to a Pod Security level. If your PSPs aren't based on the PodSecurityStandards, you
```

File: content/en/docs/tasks/configure-pod-container/migrate-from-psp.md
Comment (tengqm) [inline content/en/docs/tasks/configure-pod-container/migrate-from-psp.md:218]: Do we want to use 'Baseline' or 'baseline'? Please make them consistent through out the page.

File: content/en/docs/tasks/configure-pod-container/migrate-from-psp.md
Comment (tengqm) [inline content/en/docs/tasks/configure-pod-container/migrate-from-psp.md:278]: This is the first time we are talking about "privileged service account" ... why service account is relevant here?

File: content/en/docs/tasks/configure-pod-container/migrate-from-psp.md
Comment (tengqm) [inline content/en/docs/tasks/configure-pod-container/migrate-from-psp.md:283]:
```suggestion
arises you can easily roll the change back by deleting the RoleBinding. Just make sure the
```

File: content/en/docs/tasks/configure-pod-container/migrate-from-psp.md
Comment (tengqm) [inline content/en/docs/tasks/configure-pod-container/migrate-from-psp.md:340]: By "PodSecurity", are we referring to "PodSecurityStandard" or "PodSecurityPolicy"?

File: content/en/docs/tasks/configure-pod-container/migrate-from-psp.md
Comment (sftim) [inline content/en/docs/tasks/configure-pod-container/migrate-from-psp.md:27]:
```suggestion
## Overall approach
```

(it's sort of an anti-pattern to have the page overview and then use Overview as the heading of the first subsequent section)

File: content/en/docs/tasks/configure-pod-container/migrate-from-psp.md
Comment (sftim) [inline content/en/docs/tasks/configure-pod-container/migrate-from-psp.md:85]: (nit)
```suggestion
## 2. Simplify & standardize PodSecurityPolicies {#simplify-psps}
```

File: content/en/docs/tasks/configure-pod-container/migrate-from-psp.md
Comment (sftim) [inline content/en/docs/tasks/configure-pod-container/migrate-from-psp.md:94]: This heading doesn't feel quite right - do you have any other wording you considered here?

File: content/en/docs/tasks/configure-pod-container/migrate-from-psp.md
Comment (sftim) [reply content/en/docs/tasks/configure-pod-container/migrate-from-psp.md:132]: My suggestion:
```
You can also remove the following fields, that are related to POSIX / UNIX group controls.
```

File: content/en/docs/tasks/configure-pod-container/migrate-from-psp.md
Comment (sftim) [inline content/en/docs/tasks/configure-pod-container/migrate-from-psp.md:263]:
```suggestion
### 3.d. Bypass PodSecurityPolicy for that namespace
```

File: content/en/docs/tasks/configure-pod-container/migrate-from-psp.md
Comment (sftim) [inline content/en/docs/tasks/configure-pod-container/migrate-from-psp.md:335]: For accessibility, we should set a caption.

File: content/en/docs/tasks/configure-pod-container/migrate-from-psp.md
Comment (sftim) [inline content/en/docs/tasks/configure-pod-container/migrate-from-psp.md:326]: I would put this section / table as a standalone page in the Reference section, and hyperlink to it.

(Appendices to tasks can be harder to find than standalone reference pages, because maybe the task you're performing is “find out what Pod Security Standard I should use for this namespace, the vendor's docs only mention a PodSecurityPolicy” and we don't have a task page for that).

**Review by sftim — COMMENTED** [review]
Thanks @tallclair - this is useful stuff to have before the v1.25 release

I do have some more feedback.

File: content/en/docs/concepts/policy/pod-security-policy.md
Comment (tallclair) [reply content/en/docs/concepts/policy/pod-security-policy.md:14]: No less stable, but it is deprecated and slated for removal. If someone new to PSP lands on this page and wants to set up PodSecurityPolicy on their cluster, IMO the warning would be justified.

I'm happy to defer to your expertise on this though.

File: content/en/docs/tasks/configure-pod-container/migrate-from-psp.md
Comment (tallclair) [reply content/en/docs/tasks/configure-pod-container/migrate-from-psp.md:335]: Oops, thanks.

File: content/en/docs/tasks/configure-pod-container/migrate-from-psp.md
Comment (tallclair) [reply content/en/docs/tasks/configure-pod-container/migrate-from-psp.md:340]: Updated to `Pod Security Standards`

File: content/en/docs/tasks/configure-pod-container/migrate-from-psp.md
Comment (tallclair) [reply content/en/docs/tasks/configure-pod-container/migrate-from-psp.md:446]: Yes, thanks. I forgot we punted on allowing it to be unset when `runAsUser` was non-zero.

File: content/en/docs/tasks/configure-pod-container/migrate-from-psp.md
Comment (tallclair) [reply content/en/docs/tasks/configure-pod-container/migrate-from-psp.md:565]: Whoops! Good catch.

File: content/en/docs/tasks/configure-pod-container/migrate-from-psp.md
Comment (tallclair) [reply content/en/docs/tasks/configure-pod-container/migrate-from-psp.md:536]: Yeah. I did that because otherwise the column gets too wide and messes with the rest of the table formatting... I could cram it into each cell with a line break, would that be more clear? Can I leave the metadata.annototions bit on its own line?

The other option is to put it in a separate table to allow the columns to be sized differently.

File: content/en/docs/tasks/configure-pod-container/migrate-from-psp.md
Comment (tallclair) [reply content/en/docs/tasks/configure-pod-container/migrate-from-psp.md:326]: Sounds good, I'll follow up with this.

**Comment by tallclair** [comment]
Thanks for all the feedback. I've addressed most comments, and responded on the ones I didn't.

File: content/en/docs/concepts/policy/pod-security-policy.md
Comment (sftim) [reply content/en/docs/concepts/policy/pod-security-policy.md:14]: Let's use **caution**.

File: content/en/docs/tasks/configure-pod-container/migrate-from-psp.md
Comment (sftim) [reply content/en/docs/tasks/configure-pod-container/migrate-from-psp.md:96]: Let's live with it. I can't suggest something better.

File: content/en/docs/tasks/configure-pod-container/migrate-from-psp.md
Comment (sftim) [inline content/en/docs/tasks/configure-pod-container/migrate-from-psp.md:45]: For all of these numbered headers, I would change them to be like:
```markdown
## 0. Decide whether Pod Security Admission is right for you {#decide-whether-pod-security-admission-is-right-for-you}
```
which _omits_ the numbering. That then gives it a fragment identifier which is stable if we renumber.

(It also leaves room for a future shortcode or whatever that does auto-numbered headings.)

File: content/en/docs/tasks/configure-pod-container/migrate-from-psp.md
Comment (sftim) [reply content/en/docs/tasks/configure-pod-container/migrate-from-psp.md:545]: If you want the table to have a minimum width, we're open to setting a `class` on the table, or the row and having some accompanying CSS changes (ideally, in a separate commit).

I'd also prefer to have the white space style nowrap done with a class and some CSS.

File: content/en/docs/tasks/configure-pod-container/migrate-from-psp.md
Comment (tallclair) [reply content/en/docs/tasks/configure-pod-container/migrate-from-psp.md:45]: Oh yeah, good call.

**Comment by tallclair** [comment]
- Broke out PSP fields table into separate page
- Added "no-word-break" class on tables
- Added unnumbered section anchors
@sftim PTAL

File: content/en/docs/reference/access-authn-authz/psp-to-pod-security-standards.md
Comment (sftim) [inline content/en/docs/reference/access-authn-authz/psp-to-pod-security-standards.md:232]:
```suggestion
## PodSecurityPolicy annotations

The [annotations](/docs/concepts/overview/working-with-objects/annotations/) enumerated in this
```

I'm happy to see this merged as-is (we can log a good-first-issue about fixing it).

**Review by sftim — APPROVED** [review]
/lgtm

even with a nit
Feel free to /unhold if you want this to merge as-is @tallclair
/hold

**Comment by tallclair** [comment]
/hold cancel

Addressed last comment & squashed commits.

**Comment by tengqm** [comment]
Reapply lgtm from previous reviews.
/lgtm

**Pull request merged**
